src/widgets/section/Section.tsx

```diff
@@ -0,0 +1,55 @@
+import WidgetBase from '@dojo/framework/widget-core/WidgetBase';
+import Block from '@dojo/framework/widget-core/meta/Block';
+import { Router } from '@dojo/framework/routing/Router';
+import { tsx } from '@dojo/framework/widget-core/tsx';
+import diffProperty from '@dojo/framework/widget-core/decorators/diffProperty';
+import sectionList from '../../scripts/section-list.block';
+import * as css from './Section.m.css';
+
+import Page from '../Page';
+import SectionList from './SectionList';
+
+export interface SectionParameters {
+	section: string;
+	path?: string;
+}
+
+export interface PageData {
+	name: string;
+	path: string;
+}
+
+export default class Section extends WidgetBase<SectionParameters> {
+	private _fetchSectionList() {
+		const { section } = this.properties;
+		return this.meta(Block).run(sectionList)(section);
+	}
+
+	@diffProperty('path')
+	protected pathChange() {
+		this.invalidate();
+	}
+
+	protected render() {
+		let { path } = this.properties;
+		const { section } = this.properties;
+		const pages = this._fetchSectionList();
+
+		if (path === undefined) {
+			const item = this.registry.getInjector<Router>('router');
+			if (item && pages.length > 0) {
+				path = pages[0].path;
+				const router: Router = item.injector();
+				router.setPath(path);
```

Comment:
I don't think we should be routing in a `render`, what is this trying to do? Render the first item if it is on the root path for the section?

Reply:
That is correct. I will find another way that is not in the renderer.

Reply:
I think you should be able to conditionally render depending on whether a route is an `exact` match... if it's an `exact` match then render the first page, otherwise render the child outlet. Something like this for the index check:

```ts
<Outlet id='tutorials' renderer={(matchDetails) => {
    if (matchDetails.isExact()) {
        // render the index of the menu that includes the first tutorial
    }
}} />
```